### DISCUSSION TITLE
Compress DDS cube map?

### DISCUSSION THREAD

**Discussion by evertvorster** [ImageMagick/ImageMagick#2230] [Help]
Hi there! 

I see that ImageMagick has DDS support. I have uncompressed cube maps that I would like to compress with DXT1 and write out to a new file. 
This is the command line that I used.
convert -format dds -define dds:compression=DXT1 'cmft_cubemap.dds' 'magick_cubemap.dds'

Problem is that ImageMagick is only outputting 1 of the 6 cube faces in the output file. How do I tell it to compress all of the cube faces?

**Comment by evertvorster** [comment]
If it would help, I also have the six cube faces available as separate files, but in the cube map they have to have mipmaps, and be compressed with DXT1 to be used in the game engine that they are for.

**Comment by snibgo** [comment]
What version of IM, on what platform?

I know nothing about DDS, but for those who do, please paste a link to a sample DDS file.

**Comment by evertvorster** [comment]
[cmft_cubemap.zip](https://github.com/ImageMagick/ImageMagick/files/4872444/cmft_cubemap.zip)
[magick_cubemap.zip](https://github.com/ImageMagick/ImageMagick/files/4872447/magick_cubemap.zip)

Attached are the two files. DDS files are very compressable, as they prioritize being optimized for reading into video card texture memory over on-disk size. 

Please unzip them on your local machine. A modern GIMP can open these files quite handily if you want to see what is in them. 

The original, cmft_cubemap.dds, contains 66 layers, consisting of 6 cube faces, and 10 mipmaps at various resolutions for each of the cube faces. 
The "magic_cubemap.dds" is the output of the ImageMagic command listed in the first post. When you open the file with GIMP, you can see that only one cube face and it's mipmaps have been compressed, and the other 5 are missing.

The OS here is Arch Linux, which is a rolling release type linux OS, and so never more than a couple of weeks out of date. 

ImageMagic version is:
Version: ImageMagick 7.0.10-22 Q16 x86_64 2020-06-27
